Microsoft.Bcl.AsyncInterfaces is missing API documentation

See https://github.com/dotnet/runtime/pull/84917 for more context.

The compiler generated XML file for Microsoft.Bcl.AsyncInterfaces is shipping to customers via the nuget package and is missing public API documentation. The above linked PR disables the compiler error CS1591 until all the API is documented.

When working on this, remove the NoWarn=1591 setting from the project file.

Here are the problems:
```
C:\src\dotnet\runtime\src\libraries\System.Private.CoreLib\src\System\Runtime\CompilerServices\ConfiguredAsyncDisposable.cs(21,45): error CS1591: Missing XML comment for publicly visible type or member 'ConfiguredAsyncDisposable.DisposeAsync()' [C:\src\dotnet\runtime\src\libraries\Microsoft.Bcl.AsyncInterfaces\src\Microsoft.Bcl.AsyncInterfaces.csproj::TargetFramework=netstandard2.0]
C:\src\dotnet\runtime\src\libraries\System.Private.CoreLib\src\System\Runtime\CompilerServices\ConfiguredCancelableAsyncEnumerable.cs(39,27): error CS1591: Missing XML comment for publicly visible type or member 'ConfiguredCancelableAsyncEnumerable<T>.GetAsyncEnumerator()' [C:\src\dotnet\runtime\src\libraries\Microsoft.Bcl.AsyncInterfaces\src\Microsoft.Bcl.AsyncInterfaces.csproj::TargetFramework=netstandard2.0]
C:\src\dotnet\runtime\src\libraries\System.Private.CoreLib\src\System\Runtime\CompilerServices\EnumeratorCancellationAttribute.cs(7,25): error CS1591: Missing XML comment for publicly visible type or member 'EnumeratorCancellationAttribute' [C:\src\dotnet\runtime\src\libraries\Microsoft.Bcl.AsyncInterfaces\src\Microsoft.Bcl.AsyncInterfaces.csproj::TargetFramework=netstandard2.0]
C:\src\dotnet\runtime\src\libraries\System.Private.CoreLib\src\System\Runtime\CompilerServices\EnumeratorCancellationAttribute.cs(9,16): error CS1591: Missing XML comment for publicly visible type or member 'EnumeratorCancellationAttribute.EnumeratorCancellationAttribute()' [C:\src\dotnet\runtime\src\libraries\Microsoft.Bcl.AsyncInterfaces\src\Microsoft.Bcl.AsyncInterfaces.csproj::TargetFramework=netstandard2.0]
```

So I'd put this on @dotnet/area-system-runtime-compilerservices

--- a/src/libraries/Microsoft.Bcl.AsyncInterfaces/src/Microsoft.Bcl.AsyncInterfaces.csproj
+++ b/src/libraries/Microsoft.Bcl.AsyncInterfaces/src/Microsoft.Bcl.AsyncInterfaces.csproj
@@ -1,8 +1,6 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <TargetFrameworks>netstandard2.0;$(NetFrameworkMinimum);netstandard2.1</TargetFrameworks>
-    <!-- Public API not fully documented. -->
-    <NoWarn>$(NoWarn);1591</NoWarn>
     <IsPackable>true</IsPackable>
     <!-- This assembly should never be placed inbox as it is only for downlevel compatibility. -->
     <PackageDescription>Provides the IAsyncEnumerable&lt;T&gt; and IAsyncDisposable interfaces and helper types for .NET Standard 2.0. This package is not required starting with .NET Standard 2.1 and .NET Core 3.0.

--- a/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/ConfiguredAsyncDisposable.cs
+++ b/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/ConfiguredAsyncDisposable.cs
@@ -18,6 +18,8 @@ namespace System.Runtime.CompilerServices
             _continueOnCapturedContext = continueOnCapturedContext;
         }
 
+        /// <summary>Asynchronously releases the unmanaged resources used by the <see cref="T:System.Runtime.CompilerServices.ConfiguredAsyncDisposable" />.</summary>
+        /// <returns>A task that represents the asynchronous dispose operation.</returns>
         public ConfiguredValueTaskAwaitable DisposeAsync() =>
             // as with other "configured" awaitable-related type in CompilerServices, we don't null check to defend against
             // misuse like `default(ConfiguredAsyncDisposable).DisposeAsync()`, which will null ref by design.

--- a/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/ConfiguredCancelableAsyncEnumerable.cs
+++ b/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/ConfiguredCancelableAsyncEnumerable.cs
@@ -36,6 +36,8 @@ namespace System.Runtime.CompilerServices
         public ConfiguredCancelableAsyncEnumerable<T> WithCancellation(CancellationToken cancellationToken) =>
             new ConfiguredCancelableAsyncEnumerable<T>(_enumerable, _continueOnCapturedContext, cancellationToken);
 
+        /// <summary>Returns an enumerator that iterates asynchronously through collections that enables cancelable iteration and configured awaits.</summary>
+        /// <returns>An enumerator for the <see cref="T:System.Runtime.CompilerServices.ConfiguredCancelableAsyncEnumerable`1" /> class.</returns>
         public Enumerator GetAsyncEnumerator() =>
             // as with other "configured" awaitable-related type in CompilerServices, we don't null check to defend against
             // misuse like `default(ConfiguredCancelableAsyncEnumerable<T>).GetAsyncEnumerator()`, which will null ref by design.

--- a/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/EnumeratorCancellationAttribute.cs
+++ b/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/EnumeratorCancellationAttribute.cs
@@ -3,9 +3,11 @@
 
 namespace System.Runtime.CompilerServices
 {
+    /// <summary>Allows users of async-enumerable methods to mark the parameter that should receive the cancellation token value from <see cref="M:System.Collections.Generic.IAsyncEnumerable`1.GetAsyncEnumerator(System.Threading.CancellationToken)" />.</summary>
     [AttributeUsage(AttributeTargets.Parameter, Inherited = false)]
     public sealed class EnumeratorCancellationAttribute : Attribute
     {
+        /// <summary>Initializes a new instance of the <see cref="T:System.Runtime.CompilerServices.EnumeratorCancellationAttribute" /> class.</summary>
         public EnumeratorCancellationAttribute()
         {
         }

